### PR TITLE
Show the campus in the search subtext and add short names for buildings etc.

### PR DESCRIPTION
**Pull request by octycs**
This PR adds the campus to the search result subtext. Because this would produce too long subtexts sometimes, it also introduces short names for them.

The main motivation is that probably not many know the names of buildings like "Bestelmeyer", so it's probably helpful to show the campus as well.

Screenshot:
image

### CODE REVIEW

File: data/sources/00_areatree [hunk 1]
```diff
@@ -28,12 +28,12 @@
       0108:E-Technik Verfügungsgebäude (N8):0108,n8
       0109:Reflexionsarmer Raum (N9):0109,n9
     02:Südgelände:suedgelaende
-      0201:StudiTUM Innenstadt (S1):0201,studitum-innenstadt
+      0201:StudiTUM Innenstadt (S1)|StudiTUM:0201,studitum-innenstadt
```
Comment (CommanderStorm) [inline data/sources/00_areatree:31]: As a clarification on how the last part is hadled (I dont want to look through the parser):
what does `studitum-innenstadt` in this case mean? https://nav.tum.sexy/building/studitum-innenstadt is not availiable..

File: server/src/core/search/search_executor/postprocess.rs [hunk 2]
```diff
@@ -232,3 +229,18 @@ fn parse_room_formats(
         None
     }
 }
+
+fn generate_subtext(
+    hit: &meilisearch::MSHit,
+) -> String {
```
Comment (CommanderStorm) [inline server/src/core/search/search_executor/postprocess.rs:235]: I have not tested, if rustfmt likes this, but I assume it does
```suggestion
fn generate_subtext(hit: &meilisearch::MSHit) -> String {
```

**Review by CommanderStorm — CHANGES_REQUESTED** [review]
Overall I think this is a good adition, but I think, that a few more buildings (Raitenhaslach, Berchtesgarden, ...) from the areatree could use the shortnames.
The other comments are minor

File: data/processors/areatree.py [hunk 3]
```diff
@@ -48,7 +48,12 @@ def _parse_areatree_line(line, parent_stack, data):
         building_data["id"] = b_id
 
     # name
-    building_data["name"] = parts[1]
+    names = parts[1].split("|")
+    building_data["name"] = names[0]
```
[the remark this answers is not in the document]
Comment (octycs) [reply data/processors/areatree.py:55]: I've added a warning if it is longer than 20 chars

File: data/sources/00_areatree (hunk 1 above)
[the remark this answers is not in the document]
Comment (octycs) [reply data/sources/00_areatree:32]: It's not implemented but the idea was that there can be additional, more human readable ids that are only used for the URLs

[the remark this answers is not in the document]
Comment (CommanderStorm) [reply data/sources/00_areatree:32]: Could you write an issue for this? 😉
I can do it if you'd like..

**Review by CommanderStorm — APPROVED** [review]

File: data/sources/00_areatree (hunk 1 above)
[the remark this answers is not in the document]
Comment (octycs) [reply data/sources/00_areatree:32]: Created it: #199

**Pull request merged**